security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -18,576 +18,167 @@ package sds
-	// Same value defined in pilot pkg(k8sSAJwtTokenHeaderKey)
-	k8sSAJwtTokenHeaderKey = "istio_sds_credentials_header-bin"
 )
 
 var (
-	sdsClients       = map[cache.ConnKey]*sdsConnection{}
-	staledClientKeys = map[cache.ConnKey]bool{}
-	sdsClientsMutex  sync.RWMutex
-
-	// Tracks connections, increment on each new connection.
-	connectionNumber = int64(0)
-	sdsServiceLog    = log.RegisterScope("sds", "SDS service debugging", 0)
+	sdsServiceLog = log.RegisterScope("sds", "SDS service debugging", 0)
 )
 
-// sdsEvent represents a secret event that results in a push.
-type sdsEvent struct{}
-
-type sdsConnection struct {
-	// Time of connection, for debugging.
-	Connect time.Time
-
-	// The ID of proxy from which the connection comes from.
-	proxyID string
-
-	// The ResourceName of the SDS request.
-	ResourceName string
-
-	// Sending on this channel results in  push.
-	pushChannel chan *sdsEvent
-
-	// SDS streams implement this interface.
-	stream xds.DiscoveryStream
-
-	// The secret associated with the proxy.
-	secret *security.SecretItem
-
-	// Mutex to protect read/write to this connection
-	// TODO(JimmyCYJ): Move all read/write into member function with lock protection to avoid race condition.
-	mutex sync.RWMutex
-
-	// ConID is the connection identifier, used as a key in the connection table.
-	// Currently based on the node name and a counter.
-	conID string
-
-	// Time of the recent SDS push. Will be reset to zero when a new SDS request is received. A
-	// non-zero time indicates that the connection is waiting for SDS request.
-	sdsPushTime time.Time
-
-	// Envoy may request different versions of configuration (XDS v2 vs v3). While internally we will
-	// only generate one version or the other, because the protos are wire compatible we can cast to the
-	// requested version. This struct keeps track of the Secret type requested.
-	// Note: while SDS is entirely local, Pilot determines the version during LDS, so we need to support both versions
-	// In practice, if we are matched with a proper same version Pilot, we will always get v3.
-	secretType string
-}
-
 type sdsservice struct {
 	st security.SecretManager
 
-	ticker         *time.Ticker
-	tickerInterval time.Duration
-
-	// close channel.
-	closing chan bool
-
-	// skipToken indicates whether token is required.
-	skipToken bool
-
-	fileMountedCertsOnly bool
-
-	localJWT bool
-
-	jwtPath string
-
-	outputKeyCertToDir string
+	XdsServer *xds.DiscoveryServer
+	stop      chan struct{}
+}
 
-	// Credential fetcher
-	credFetcher security.CredFetcher
+func NewXdsServer(stop chan struct{}, gen model.XdsResourceGenerator) *xds.DiscoveryServer {
+	s := xds.NewXDS(stop)
+	s.DiscoveryServer.Generators = map[string]model.XdsResourceGenerator{
+		v3.SecretType: gen,
+	}
+	s.DiscoveryServer.ProxyNeedsPush = func(proxy *model.Proxy, req *model.PushRequest) bool {
+		found := false
```

Comment:
Could you briefly explain the logic in this ProxyNeedsPush and add comment here?